src/lib/y2network/sysconfig/connection_config_readers/ethernet.rb

```diff
@@ -17,31 +17,22 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
+require "y2network/sysconfig/connection_config_readers/base"
 require "y2network/connection_config/ethernet"
 
 module Y2Network
   module Sysconfig
     module ConnectionConfigReaders
       # This class is able to build a ConnectionConfig::Ethernet object given a
       # Sysconfig::InterfaceFile object.
-      class Ethernet
-        # @return [Y2Network::Sysconfig::InterfaceFile]
-        attr_reader :file
-
-        # Constructor
-        #
-        # @param file [Y2Network::Sysconfig::InterfaceFile] File to get interface configuration from
-        def initialize(file)
-          @file = file
-        end
-
+      class Ethernet < Base
         # @return [Y2Network::ConnectionConfig::Ethernet]
         def connection_config
           Y2Network::ConnectionConfig::Ethernet.new.tap do |conn|
             conn.bootproto = file.bootproto
             conn.description = file.name
             conn.interface = file.interface
-            conn.ip_address = file.ip_address
+            conn.ip_configs = ip_configs
```

Comment:
hmm, considering  that in future maybe more shared stuff will be there. what about protected method that take conn and file and assign it?

I expect it can already read bootproto, description, interface and those ip_configs. as it is probably for all interfaces.

Reply:
ip_configs is already a method (from `Base` class) and the `file` is available as an instance variable.

Reply:
this is actually not what I mean. I mean that you repeat in all specialized class stuff like
```
conn.bootproto = BootProtocol.from_name(file.bootproto || "static")
￼conn.description = file.name
conn.interface = file.interface
conn.ip_configs = ip_configs
conn.startmode = Startmode.create(file.startmode || "manual")
conn.startmode.priority = file.ifplugd_priority if conn.startmode.name == "ifplugd"
```

so I expect it should be in that base class and onlly class specific stuff in this config constructor.